templates/components/sidebar.html

```diff
@@ -1 +1,26 @@
-<div id="app-sidebar"></div>
\ No newline at end of file
+<div id="app-sidebar">
```

Comment:
Why are you adding these at that place? I think it's better to avoid bringing all of them here for a number of reason (they might be removed by some installations, or even the URLs changed etc)

Reply:
Mainly because that's where they are generated in the frontend, so when they're replaced during load, the markup is similar.  This means it's compatible with the existing stylesheet to prevent it from blinking and rearranging as the page progressively loads on slower connections.  The differences from the final frontend-generated version also mean that any search engines indexing the page which do partially evaluate JavaScript (such as Google) don't ding you for presenting something different to users than what the search engine sees.